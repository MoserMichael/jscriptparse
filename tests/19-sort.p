# test sort with a callback

def cmp(x, y) {
    if x[1] < y[1] return -1
    if x[1] > y[1] return 1
    return 0
}
r=sort([['a',100],['b',1],['c',1000]],cmp) 
println(toJsonString(r))

# test sort without a callback

r=sort(range(10,1,-1))
println(toJsonString(r))

