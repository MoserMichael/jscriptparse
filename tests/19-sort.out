[["b",1],["a",100],["c",1000]]
[2,3,4,5,6,7,8,9,10]
